File: Advanced/Arrays/Fundamentals/2_largest_element.cpp
#include <bits/stdc++.h>
using namespace std;
// Largest Element
// TC = O(n)
// SC = O(1)
int largestElement(vector<int> &nums)
{
    int n = nums.size();
    int max = nums[0];
    for (int i = 0; i < n; i++)
    {
        if (max < nums[i])
            max = nums[i];
    }
    return max;
}
int main()
{
    vector<int> nums = {3, 2, 1, 5, 2};
    cout << largestElement(nums) << endl;
    return 0;
}

File: Advanced/Arrays/Fundamentals/2_largest_element.py
# Largest Element
# TC = O(n)
# SC = O(1)
def largest_element(nums):
    if not nums:
        return None
    largest = nums[0]
    for num in nums:
        if num > largest:
            largest = num
    return largest


nums = [3, 1, 4, 1, 5, 9, 2, 6, 5]
result = largest_element(nums)
print(result)
